astropy/io/fits/fitsrec.py

```diff
@@ -1283,19 +1284,50 @@ def _get_recarray_field(array, key):
     return field
 
 
-def _rstrip_inplace(array, chars=None):
+def _rstrip_inplace(array):
     """
-    Performs an in-place rstrip operation on string arrays.
-    This is necessary since the built-in `np.char.rstrip` in Numpy does not
-    perform an in-place calculation.  This can be removed if ever
-    https://github.com/numpy/numpy/issues/6303 is implemented (however, for
-    the purposes of this module the only in-place vectorized string functions
-    we need are rstrip and encode).
+    Performs an in-place rstrip operation on string arrays. This is necessary
+    since the built-in `np.char.rstrip` in Numpy does not perform an in-place
+    calculation.
     """
 
-    for item in np.nditer(array, flags=['zerosize_ok'],
-                                 op_flags=['readwrite']):
-        item[...] = item.item().rstrip(chars)
+    # The following implementation convert the string to unsigned integers of
+    # the right length. Trailing spaces (which are represented as 32) are then
+    # converted to null characters (represented as zeros). To avoid creating
+    # large temporary mask arrays, we loop over chunks (attempting to do that
+    # on a 1-D version of the array; large memomry may still be needed in the
+    # unlikely case that a string array has small first dimension and cannot
+    # be represented as a contiguous 1-D array in memory).
+
+    dt = array.dtype
+
+    if dt.kind not in 'SU':
+        raise TypeError("This function can only be used on string arrays")
+    bpc = 1 if dt.kind == 'S' else 4
+    dt_int = "{0}{1}u{2}".format(dt.itemsize // bpc, dt.byteorder, dt.bpc)
+    b = np.array(array, copy=False).view(dt_int)
+    # For optimal speed, work in chunks of the internal ufunc buffer size.
+    bufsize = np.getbufsize()
+    # Attempt to make it a 1-D array so that the chunks have known size.
+    # We cannot use ravel, since that will copy for non-contiguous arrays.
+    # and we need a view of the input array. The code will work on non-1D
+    # arrays, although the chunks will now be larger.
+    if b.ndim > 1:
+        try:
+            b.shape = -1
```

Comment:
Hmm, it took me some time to understand the previous version, but now I'm lost: if you flatten the array how do you know the beginning/end of each string ? and the chunking below may split a string ? After some testing it does not seem to work, though it works without this.

Reply:
Dud, that's stupid of me!!